src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java

```
@@ -148,4 +156,65 @@ public String getCmekKey(Arguments arguments) {
     }
     return arguments.get("gcp.cmek.key.name");
   }
+
+  public CryptoKeyName parseCmekKey(String key, FailureCollector collector) {
+    CryptoKeyName cmekKeyName = null;
+    try {
+      cmekKeyName = CryptoKeyName.parse(key);
+    } catch (ValidationException e) {
+      collector.addFailure(e.getMessage(), null)
+        .withConfigProperty(NAME_CMEK_KEY).withStacktrace(e.getStackTrace());
+    }
+    return cmekKeyName;
+  }
+
+  public boolean requiredFieldsContainsMacro() {
+    return containsMacro(NAME_PROJECT) || containsMacro(NAME_SERVICE_ACCOUNT_TYPE) ||
+      containsMacro(NAME_SERVICE_ACCOUNT_JSON) || containsMacro(NAME_SERVICE_ACCOUNT_FILE_PATH);
+  }
+
+  public Credentials getCredentials() {
+    Boolean isServiceAccountFilePath = isServiceAccountFilePath();
+    Credentials credentials = null;
+    try {
+      credentials = getServiceAccount() == null ?
+        null : GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath);
+    } catch (Exception e) {
+      /* Ignoring the exception because we don't want to highlight config if an exception occurs while
```

this is not appropriate for a utility method, correct handling of the exception depends on what you're trying to do. For example, at pipeline runtime you would never want to ignore the exception.

---

Yeah, so I looked how it is done in existing code and made the changes accordingly.